### PR TITLE
Feat:-Add summary component instances

### PR DESCRIPTION
**Pull request by Biki-das**
Its just a small POC and if it gets some good response and is actually useful then makes sense
Currently we have our tree View in the React Devtools, rendering the components instances in a tree structure 

How about a new tab which sort of acts like a summary  of components, like showing the number of instances of a particular component is mounted at a particular render cycle.

thoughts @hoxyq 

https://github.com/facebook/react/assets/72331432/628e3c2c-2f0f-4372-b9a0-717e7f5c7532

we can do some sort of vote in twitter or somewhere what people think about this

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/store.js [hunk 1]
```diff
@@ -1444,6 +1444,12 @@ export default class Store extends EventEmitter<{
     copy(text);
   };
 
+  getAllElements = (): Array<{id: number, displayName: string | null}> => {
```
Comment (Biki-das) [inline packages/react-devtools-shared/src/devtools/store.js:1447]: kinda clueless about the flow error!, things are working fine, need some help here

**Review by vvvvvzc — COMMENTED** [review]
fff

**Comment by hoxyq** [comment]
Hey, I will get to it probably at the start of the next week, currently fixing potential next release.

Thanks for working on this! Let's think how we can use this data:
- What is the main purpose for this? What this data can show me as a developer?
- Maybe we can show this data in a different way? Like a piechart or something. Not important for now, just thinking ahead.

**Comment by Biki-das** [comment]
> Hey, I will get to it probably at the start of the next week, currently fixing potential next release.
> 
> Thanks for working on this! Let's think how we can use this data:
> 
> * What is the main purpose for this? What this data can show me as a developer?
> * Maybe we can show this data in a different way? Like a piechart or something. Not important for now, just thinking ahead.

I understand, get back whenever its possible for you, i can answer the main purpose as far for now it indicates the number of instances a particular component is mounted at a particular render cycle.
Probably can be used for performance analysis etc.
and thanks for the suggestions may be we can refine the idea more, lets see.

File: packages/react-devtools-shared/src/devtools/store.js (hunk 1 above)
Comment (hoxyq) [inline packages/react-devtools-shared/src/devtools/store.js:1447]:
```suggestion
  getAllElements = () => Array<{id: number, displayName: string | null}> = () => {
```

Comment (hoxyq) [reply packages/react-devtools-shared/src/devtools/store.js:1447]: This should probably work

Comment (Biki-das) [reply packages/react-devtools-shared/src/devtools/store.js:1447]: thanks @hoxyq let me try and test it.

**Comment by Biki-das** [comment]
> 
hmm fair enough, so we shall close this ?

**Pull request closed without merging**